File: api-country-code/src/main/java/com/kolyall/countryapi/objects/Country.java
package com.kolyall.countryapi.objects;

import android.support.annotation.NonNull;

import com.google.gson.annotations.Expose;
import com.kolyall.countryapi.utils.CountryUtils;

import java.text.Collator;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


/**
 * Created by hbb20 on 11/1/16.
 */
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter @Setter
public class Country implements Comparable<Country> {
    @Expose String regionCode;
    @Expose String countryCode;
    @Expose String name;
    @Expose String englishName;

    @Builder
    public Country(String nameCode, String phoneCode, String name) {
        this(nameCode,phoneCode,name,name);
    }

    public int getFlagDrawableResId() {
        return CountryUtils.getFlagDrawableResId(this);
    }

    /**
     * If country have query word in name or name code or phone code, this will return true.
     */
    public boolean isEligibleForQuery(String query) {
        query = query.toLowerCase();
        return getName().toLowerCase().contains(query) || getRegionCode().toLowerCase().contains(query) || getCountryCode().toLowerCase().contains(query) || getEnglishName().toLowerCase().contains(query);
    }

    @Override
    public int compareTo(@NonNull Country other) {
        return Collator.getInstance().compare(this.getName(), other.getName());
    }
}

File: api-country-code/src/main/java/com/kolyall/countryapi/objects/Country.kt
package com.kolyall.countryapi.objects

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.kolyall.countryapi.utils.CountryUtils
import com.kolyall.countryapi.utils.flagDrawableResId
import java.text.Collator

/**
 * Created by hbb20 on 11/1/16.
 */
class Country(
    @SerializedName("region_code")
    @Expose
    var regionCode: String = "",

    @SerializedName("country_code")
    @Expose
    var countryCode: String = "",

    @SerializedName("english_name")
    @Expose
    var englishName: String = ""
) : Comparable<Country> {

    @SerializedName("name")
    @Expose
    lateinit var name: String

    /**
     * If country have query word in name or name code or phone code, this will return true.
     */
    fun isEligibleForQuery(text: String): Boolean {
        val query = text.toLowerCase()
        return name.toLowerCase().contains(query)
            || regionCode.toLowerCase().contains(query)
            || countryCode.toLowerCase().contains(query)
            || englishName.toLowerCase().contains(query)
    }

    override fun compareTo(other: Country): Int {
        return Collator.getInstance().compare(name, other.name)
    }
}